chore!: remove `Opcode::Brillig` from ACIR

TomAFrench opened the pull request:
This PR removes `Opcode::Brillig` as it has been superceded by `Opcode::BrilligCall`

TomAFrench committed:
chore!: remove `Opcode::Brillig` from ACIR

TomAFrench committed:
chore: fix clippy

TomAFrench committed:
chore: update acir in acvm_js test

TomAFrench commented:
I've cut some corners related to the ACVM's interface as this is handled in https://github.com/noir-lang/noir/pull/4897.

TomAFrench committed:
Merge branch 'master' into tf/remove-brillig

ludamad committed:
chore(ci): back to on-demand (#5998)

We were seeing spot reaps, if costs are not terrible we should consider
on-demand

ledwards2225 committed:
chore: Clean up and clarify some translator flavor logic (#5965)

Cleans up Translator flavor: removes some redundant methods, replaces
list style getters with one built up from sub-components, and clarifies
the subtlety around getters that return everything except concatenated
polys.

TomAFrench commented:
This is now ready go in but we're going to do another sync between repos before merging.

jfecher commented on noir/noir-repo/tooling/debugger/src/context.rs:
```diff
@@ -551,16 +551,16 @@ impl<'a, B: BlackBoxFunctionSolver> DebugContext<'a, B> {
-                        }
-                    }
+                    && matches!(opcodes[acir_index], Opcode::BrilligCall { .. })
+                // && {
+                //     if let Opcode::Brillig(ref brillig) = opcodes[acir_index] {
+                //         brillig_index < brillig.bytecode.len()
+                //     } else {
+                //         false
+                //     }
+                // }
```
```suggestion
```
If these snippets are a TODO for remaining work we should make that more explicit

jfecher commented on noir/noir-repo/tooling/debugger/src/repl.rs:
```diff
@@ -153,21 +153,23 @@ impl<'a, B: BlackBoxFunctionSolver> ReplDebugger<'a, B> {
-                .is_breakpoint_set(&OpcodeLocation::Brillig { acir_index, brillig_index })
-            {
-                " *"
+            // } else if self
+            //     .context
+            //     .is_breakpoint_set(&OpcodeLocation::Brillig { acir_index, brillig_index })
+            // {
+            //     " *"
```
TODO could be used here as well

jfecher reviewed (commented):
Can you mark the remaining work a bit more explicitly with TODOs or similar? They look like they'd be hard to find after the fact at the moment.

Edit: Cross-referenced the debugger PR and resolved my comments related to unimplemented work from there. I originally thought we'd come back later to fix it so TODOs would be useful but if we already have the code then it should be fine.

jfecher commented on noir/noir-repo/tooling/debugger/src/context.rs:
```diff
@@ -296,20 +296,20 @@ impl<'a, B: BlackBoxFunctionSolver> DebugContext<'a, B> {
-                } else {
-                    format!("{opcode:?}")
-                }
+                // if let Opcode::Brillig(ref brillig) = opcode {
+                //     let first_opcode = &brillig.bytecode[0];
+                //     format!("BRILLIG {first_opcode:?}")
+                // } else {
+                format!("{opcode:?}")
+                // }
```
I haven't checked the debugger PR but if it would just merge over this one then we can ignore these

spypsy committed:
fix: make discv5 test deterministic (#5968)

vezenovm commented on noir/noir-repo/acvm-repo/acvm/src/pwg/mod.rs:
```diff
@@ -469,41 +423,11 @@ impl<'a, B: BlackBoxFunctionSolver> ACVM<'a, B> {
-        }
-        self.brillig_solver = Some(solver);
-        self.solve_opcode()
+    pub fn finish_brillig_with_solver(&mut self, _solver: BrilligSolver<'a, B>) -> ACVMStatus {
+        unimplemented!("Opcode::Brillig is removed")
```
They are needed for the debugger, let's link to those issues though (https://github.com/noir-lang/noir/issues/4823)

AztecBot committed:
feat: Sync from noir (#5999)

Automated pull of development from the
[noir](https://github.com/noir-lang/noir) programming language, a
dependency of Aztec.
BEGIN_COMMIT_OVERRIDE
fix: fix curve parameters for bigints
(https://github.com/noir-lang/noir/pull/4900)
chore: bump `rustls` to v0.21.11
(https://github.com/noir-lang/noir/pull/4895)
fix: Update noir-gates-diff commit to use master reference report
(https://github.com/noir-lang/noir/pull/4891)
fix: Reset the noir-gates-diff report on master
(https://github.com/noir-lang/noir/pull/4878)
fix(experimental): Skip over comptime functions in scan pass
(https://github.com/noir-lang/noir/pull/4893)
chore(experimental): Improve variable not defined error message in
comptime interpreter (https://github.com/noir-lang/noir/pull/4889)
chore(experimental): Add scan pass and `into_expression` for comptime
interpreter (https://github.com/noir-lang/noir/pull/4884)
END_COMMIT_OVERRIDE

---------

Co-authored-by: Tom French <tom@tomfren.ch>

vezenovm committed:
chore: Reset noir-gates-diff report on master  (#6003)

Similarly to https://github.com/noir-lang/noir/pull/4878 the gates diff
report got corrupted following changes to `nargo info`. This PR simply
resets the gates diff so it is expected to see really large gates
differences. You can look into the Noir PR for more info. There will be
a followup which sets back to the correct noir-gates-diff commit.
Without this reset we will not be able to get accurate gate diffs based
off of master.

vezenovm committed:
fix: Use correct gates diff commit now that master has been reset (#6004)

Follow-up to https://github.com/AztecProtocol/aztec-packages/pull/6003.
This PR simply updates to using a noir gates diff that expects an
uncorrupted comparison report on master.

ludamad committed:
chore(ci): hotfix runners not starting

alexghr committed:
fix: refuse to start sequencer without a prover (#6000)

Avoids a bad situation where a sequencer node produces empty blocks that fail to validate in the rollup contract.

alexghr committed:
feat: serialize public kernel private inputs (#5971)

Serialize `PublicKernelCircuitPrivateInputs` to and from strings

LHerskind committed:
feat: add the storage layout to the contract artifact (#5952)

Fixes #5947 and includes the notes as well. The notes was part of the
contract object, but not directly exposed on the artifact itself.

Needed to fix a few additional things to make it work:
- The `deploy_contract` function was broken (old call flow), it cannot
do the new flow because that would be a circular dependency.
- Using the `storageLayout` values to get the `storageSlot` that we are
using in multiple tests
- Using the `notes` values to get the `noteTypeId` that we are using in
multiple tests
- Removed `lodash.uniqby` as the dependency was not needed after this.

ludamad committed:
hotfix: better docker prune

vezenovm commented:
> This is now ready go in but we're going to do another sync between repos before merging.

Maybe we should wait for https://github.com/noir-lang/noir/pull/4897 to be merged and syncd with this repo before merging this PR

benesjan committed:
feat: AES oracle (#5996)

Fixes #5895

AztecBot committed:
feat: Sync from noir (#6007)

Automated pull of development from the
[noir](https://github.com/noir-lang/noir) programming language, a
dependency of Aztec.
BEGIN_COMMIT_OVERRIDE
fix: Nested array equality (https://github.com/noir-lang/noir/pull/4903)
feat: Handle `BrilligCall` opcodes in the debugger
(https://github.com/noir-lang/noir/pull/4897)
chore: Release Noir(0.28.0)
(https://github.com/noir-lang/noir/pull/4776)
feat: Sync from aztec-packages
(https://github.com/noir-lang/noir/pull/4902)
END_COMMIT_OVERRIDE

---------

Co-authored-by: Tom French <tom@tomfren.ch>
Co-authored-by: Tom French <15848336+TomAFrench@users.noreply.github.com>

TomAFrench committed:
Merge branch 'master' into tf/remove-brillig

* master:
  feat: Sync from noir (#6007)
  feat: AES oracle (#5996)
  hotfix: better docker prune
  feat: add the storage layout to the contract artifact (#5952)
  feat: serialize public kernel private inputs (#5971)
  fix: refuse to start sequencer without a prover (#6000)
  chore(ci): hotfix runners not starting
  fix: Use correct gates diff commit now that master has been reset (#6004)
  chore: Reset noir-gates-diff report on master  (#6003)
  feat: Sync from noir (#5999)
  fix: make discv5 test deterministic (#5968)
  chore: Clean up and clarify some translator flavor logic (#5965)
  chore(ci): back to on-demand (#5998)

TomAFrench committed:
chore: revert stuff

TomAFrench committed:
chore: delete dead code

TomAFrench committed:
chore: revert change

jfecher reviewed (approved):
👍
